Neos.Media/Classes/Domain/Model/ThumbnailGenerator/DocumentThumbnailGenerator.php

```diff
@@ -116,4 +117,26 @@ public function refresh(Thumbnail $thumbnail)
             throw new Exception\NoThumbnailAvailableException($message, 1433109652, $exception);
         }
     }
+
+    private static function originalAssetResourceIsNotEncrypted(Thumbnail $thumbnail): bool
+    {
+        $extension = $thumbnail->getOriginalAsset()->getResource()->getFileExtension();
+        if ($extension !== 'pdf') {
+            return true;
+        }
+        $stream = $thumbnail->getOriginalAsset()->getResource()->getStream();
+        if ($stream === false) {
+            return true;
+        }
+
+        while (($chunk = fgets($stream, 1024)) !== false) {
+            if (strpos($chunk, '/Encrypt') !== false) {
+                fclose($stream);
+                return false;
+            }
+        }
+        fclose($stream);
```

Comment:
Hmm, I don't like that all non-encrypted PDF get fully read to discover, that they are not encrypted (99.9% of the pdf I guess). Why do we not provide a default thumbnail in general for documents, where the thumbnail creation is broken for any reason?

Reply:
It won't read the complete file - in fact it will stop as soon as it finds "/Encrypt" going in 1kb chunks over the file - which usually should hit with the first chunk. (the `fclose` will bring the `fgets` read in the loop header to fail)

Reply:
Ahhhh... but it will read unencrypted PDFs - as you wrote. 

And yep - there again is room for improvement, I did not "detect" - you are right!